tests/shouldwork/Issues/T1171.hs

```diff
@@ -0,0 +1,46 @@
+-- Test that enable lines are treated specially by Clash, meaning they have
+-- the type "main_types.en_system" in the generated HDL.
+--
+module T1171 where
+
+import Prelude as P
+
+import Data.List
+import System.Environment
+import System.FilePath
+
+import Clash.Prelude
+
+topEntity
+  :: (HiddenClockResetEnable System)
+  => Signal System (BitVector 4)
+  -> Signal System Bool
+topEntity = fmap (\x -> msb x == lsb x)
+
+assertIn :: String -> String -> IO ()
+assertIn needle haystack
+  | needle `isInfixOf` haystack = return ()
+  | otherwise = P.error $ mconcat [ "Expected:\n\n  ", needle
+                                  , "\n\nIn:\n\n", haystack ]
+
+mainVHDL :: IO ()
+mainVHDL = do
+  [topDir] <- getArgs
+  content  <- readFile (takeDirectory topDir </> "topentity.vhdl")
+
+  assertIn "\\c$$d(%,,%)_2\\ : in t1171_types.en_system;" content
+
+mainVerilog :: IO ()
+mainVerilog = do
+  [topDir] <- getArgs
+  content  <- readFile (takeDirectory topDir </> "topEntity.v")
+
+  assertIn "input  \\c$$d(%,,%)_2  // enable" content
+
+mainSystemVerilog :: IO ()
+mainSystemVerilog = do
+  [topDir] <- getArgs
+  content  <- readFile (takeDirectory topDir </> "topEntity.sv")
+
+  assertIn "input logic \\c$$d(%,,%)_2   // enable" content
```

Comment:
I would leave out the weird generated names from the patterns you're looking for.
Or better yet, use [PortNames](https://hackage.haskell.org/package/clash-prelude-1.2.1/docs/Clash-Annotations-TopEntity.html#t:PortName) to to give the enable port an explicit name.

Reply:
`PortName` is a better idea IMO, just looking for the comment `// enable` seems a little too lax for my taste